lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/services/LibertyWorkspace.java

```diff
@@ -89,7 +91,7 @@ public void setInstalledFeatureList(List<Feature> installedFeatureList) {
     public void initConfigFileList() {
         try {
             List<Path> configPathsList = Files.find(Paths.get(getWorkspaceURI()), Integer.MAX_VALUE, (filePath, fileAttributes) -> 
-                    LibertyUtils.isServerXMLFile(filePath.toString()) || (fileAttributes.isDirectory() && filePath.toString().endsWith("configDropins")))
+                    LibertyUtils.isServerXMLFile(filePath.toString()) || isConfigDir(filePath, fileAttributes))
```

Comment:
When the filePath is a config dir, does the dir get added to the list? What about the files within that dir? How do they get added to the list?

Reply:
When filePath is a config dir, it does not get added to the list. All the files within the dir are added with no distinction. Although, that can be changed for thoroughness.
I add them with `Files.list(configPath).forEach(path -> configFiles.add(path.toString()));` in the following `for` loop. 
The dir scan only trigger on initialization.

Reply:
The dir scan on initialization sounds right, and adding the files within at that time. If they create a new file later on in one of these config dirs, will it be recognized as a Liberty config file? If not, is there a reasonable way for our extension to figure that out?

Reply:
I realized a simpler solution to be to simply check the file path lazily to see if it contains a config dir.

Reply:
I like the new approach.